thoth/user_api/api_v1.py

```diff
@@ -241,15 +241,25 @@ def get_analyze_status(analysis_id: str) -> typing.Tuple[typing.Dict[str, typing
 
 def post_provenance_python(
     application_stack: dict,
-    origin: str = None,
     debug: bool = False,
     force: bool = False,
-    kebechet_metadata: typing.Dict[str, typing.Any] = None,
+    origin: str = None,
     justification: typing.Optional[typing.List[typing.Dict[str, typing.Any]]] = None,
     stack_info: typing.Optional[typing.List[typing.Dict[str, typing.Any]]] = None,
+    # Must be set to set protected fields
+    internal_secret: typing.Optional[str] = None,
+    # Protected fields
+    kebechet_metadata: typing.Optional[typing.Dict[str, typing.Any]] = None,
 ):
     """Check provenance for the given application stack."""
+    protected_fields = ["kebechet_metadata"]
     parameters = locals()
+
+    if Configuration.INTERNAL_SECRET == internal_secret:
+        for k in protected_fields:
+            if parameters[k] is not None:
+                raise PermissionError("Attempted to set protected field without proper permissions.")
```

Comment:
This will result in an internal error. It might be good idea to report back a JSON response with an error signalizing HTTP status code.